back/boxtribute_server/models/crud.py

```diff
@@ -191,6 +191,56 @@ def delete_tag(*, user_id, id):
     return tag
 
 
+def assign_tag(*, user_id, id, resource_id, resource_type):
+    """Create TagsRelation entry as cross reference of the tag given by ID, and the
+    given resource (a box or a beneficiary). Insert timestamp for modification in
+    resource model.
+    Validate that tag type and resource type are compatible.
+    Return the resource.
+    """
+    tag = Tag.get_by_id(id)
+    if (
```

Comment:
@pylipp can you note down for our next meeting that we should talk about input validation in the backend. 
cause you are checking here if a beneficiary type is applied to a box or the other way around, but what if there was no type passed at all or ....

Reply:
Sure! The test suite is never complete :wink: 
Note that the GraphQL API requires a non-null resource type to be passed for `TagOperationInput`.